spi/src/main/java/org/opensearch/jobscheduler/spi/schedule/CronSchedule.java

```diff
@@ -96,19 +105,24 @@ String getCronExpression() {
         return this.expression;
     }
 
+    public long getDelay() { return this.scheduleDelay; }
+
+    public void setDelay(long delay) { this.scheduleDelay = delay; }
+
     @Override
     public Instant getNextExecutionTime(Instant time) {
         Instant baseTime = time == null ? this.clock.instant() : time;
 
-        ZonedDateTime zonedDateTime = ZonedDateTime.ofInstant(baseTime, this.timezone);
+        // the executionTime object doesn't use the delay, need to remove the delay before and then add it back after
+        ZonedDateTime zonedDateTime = ZonedDateTime.ofInstant(baseTime.minusMillis(this.scheduleDelay), this.timezone);
```

Comment:
I am not sure I follow this - why do we need to remove from baseTime and add to executionTime later?

Reply:
I did it this way to isolate the delay to the CronSchedule class rather than modifying the ExecutionTime object to include the delay parameter. 
For example, say it is 10:07 AM and I have a cron schedule set up to run every hour with a delay of 15 minutes. Given an input time of 10:07 AM, the ExecutionTime object would return the next execution as 11 AM. By subtracting the delay first, the ExecutionTime object is given the input time as 9:52 AM, and it returns the next execution time of 10:00 AM. After adding the delay, we get the correct next execution time of 10:15 AM.
It is probably too confusing though, do you think it would be better to just include the delay in the ExecutionTime class to avoid all of this jumping around?